jobs/buildhub/utils.py

```diff
@@ -177,36 +198,27 @@ def is_build_url(product, url):
     ):
         return False
 
-    re_exclude = re.compile(
-        '.+(tinderbox|try-builds|partner-repacks|latest|contrib|/0\.|'
-        'experimental|namoroka|debug|sha1-installers|candidates/archived|'
-        'stylo-bindings|/1.0rc/|/releases/win../|dominspector|/test/|testing|'
-        '%28.+%29|\sInstaller\.(\w{2,3}\-?\w{0,3})\.exe)'
-    )
-    if re_exclude.match(url):
+    if _build_url_exclude_names_regex.match(url):
         return False
 
+    extensions = FILE_EXTENSIONS
     # Only .exe for Windows.
-    extensions = list(FILE_EXTENSIONS)
     if 'win' in url:
-        extensions.remove('zip')
+        extensions = [x for x in extensions if x != 'zip']
 
     if product == 'devedition':
         product = 'firefox'
     if product == 'mobile':
         product = 'fennec'
     filename = os.path.basename(url)
     match_filename = filename.replace(' ', '-').lower()
+    if _build_url_exclude_suffixes_regex.match(match_filename):
+        return False
```

Comment:
These lines used to be on the last line before. I just moved them up in case it can cause an early exit and thus we don't need to make the last regex stuff after this.